python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -406,11 +549,25 @@ def get_vocabulary_info(feature_info, file_io):
     else:
         vocabulary_keys = vocabulary_df.iloc[:, 0]
     if "max_length" in args:
-        vocabulary_keys = vocabulary_keys[:args["max_length"]]
+        vocabulary_keys = vocabulary_keys[: args["max_length"]]
     if "default_value" in feature_info:
         vocabulary_keys = vocabulary_keys.fillna(feature_info["default_value"])
     vocabulary_keys = vocabulary_keys.values
-    vocabulary_ids = (
-        vocabulary_df["id"].values if "id" in vocabulary_df else list(range(len(vocabulary_keys)))
-    )
-    return vocabulary_keys, vocabulary_ids
\ No newline at end of file
+    if "dropout_rate" in args:
+        # NOTE: If a dropout_rate is specified, then reserve 0 as the OOV index
+        vocabulary_ids = (
+            vocabulary_df["id"].values
+            if "id" in vocabulary_df
+            else list(range(1, len(vocabulary_keys) + 1))
```

Comment:
We should have:
```
OOV = 1
list(range(1, len(vocabulary_keys) + OOV))
```

Reply:
@lastmansleeping Thoughts on Guillaume's suggestion here? Would we want this 00V index customizable here?